internal/cloudapi/client_test.go

```diff
@@ -33,3 +33,15 @@ func TestOsbuildClientGetNoError(t *testing.T) {
 	require.NotNil(t, new_client)
 	require.Equal(t, "localhost:8086/", new_client.Server)
 }
+
+func TestOsbuildClientConfigureClientReturnsNilWhenNotUsingHttps(t *testing.T) {
+	// we're not using HTTPS URL on purpose
+	osbuild_client := NewOsbuildClient("localhost:8086/", nil, nil, nil)
```

Comment:
hmm tbh for this I'd generate some certificates, put them in a temporary directory. And create the client with those. Otherwise this test kinda doesn't do anything new compared to the ones above (as ConfigureClient does nothing without certs).

Reply:
this one checks the result of ConfigureClient explicitly which indeed isn't much, but that result is also used in NewClient itself. While the above test doesn't check anything related to the execution of ConfigureClient.

My next scenario would be with certificates.